### PR TITLE
Eng 1180 automatically find the next available port for the server

### PR DESCRIPTION
**Pull request by cw75**
Tested and it works.

### CODE REVIEW

File: src/python/bin/aqueduct [hunk 1]
```diff
@@ -182,11 +188,36 @@ def start(expose):
                 execute_command(["rm", os.path.join(server_directory, "__version__")])
             exit(1)
 
+    server_port = default_server_port
+    while is_port_in_use(server_port):
+        server_port += 1
+    
+    print("Next available port is %s" % server_port)
```
Comment (vsreekanti) [inline src/python/bin/aqueduct:195]: Nit -- let's say "Port 8080 is occupied; next available port is..."

Comment (vsreekanti) [reply src/python/bin/aqueduct:195]: Also, another nit: Why are we using `%s` for integers?

**Review by vsreekanti — COMMENTED** [review]
@cw75, can we add functionality for the user to explicitly specify the port if they want via a `--port` argument on the CLI?

**Comment by cw75** [comment]
@vsreekanti addressed your comments.

File: src/python/bin/aqueduct [hunk 2]
```diff
@@ -296,6 +331,7 @@ if __name__ == "__main__":
                                ''')
     start_args.add_argument('--expose', default=False, action='store_true',
                     help="Use this option to expose the server to the public.")
+    start_args.add_argument('--port', dest="port", help="Specify the port that the server listens to.")
```
Comment (vsreekanti) [inline src/python/bin/aqueduct:334]: English nit: "Specify the port on which the Aqueduct server runs."

**Review by vsreekanti — APPROVED** [review]

**Pull request merged**